week2/homework/src/index.js

```diff
@@ -1,3 +1,82 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+
+const { readFile, writeFile } = require('fs');
+const { promisify } = require('util');
+const readFileWithPromise = promisify(readFile);
+const writeFileWithPromise = promisify(writeFile);
+
+const args = process.argv.slice(2);
+
+const cmd = args[0];
+
+const TODO_PATH = 'todo.json';
+
+function readTodos() {
+    return readFileWithPromise(TODO_PATH, 'utf8')
+        .then(JSON.parse)
+        .catch(() => ([]));
+}
+
+function writeTodos(todos) {
+    return writeFileWithPromise(TODO_PATH, JSON.stringify(todos, null, 2));
+}
+
+function myHelper() {
+    console.log('Get help from these commands:1- node.add \'New todo\'   = Add a todo 2- node . delete [Delete item]  = Delete item from the list 3- node . list  = List all items 4- node . update [Item to be updated]  \'New todo\'  = Updating a todo. 5- node . reset  = Reset the list');
+}
+
+switch (cmd) {
+    case 'add':
+        const text = args[1];
+        readTodos()
+            .then(todos => {
+                todos.push({ todo: text });
+                return todos;
+            })
+            .then(writeTodos)
+            .then(readTodos)
+            .then(console.log);
+        break;
+
+    case 'delete':
+        let item = args[1];
+        readTodos()
+            .then(todos => {
+                if (item > 0) {
+                    todos.splice(item, 1);
```

Comment:
The assignment was to use a 1-based index to delete and update items so when I call `delete 1` the first item is deleted and not the second one as is with your code. Also the command was supposed to be called `remove`.

Also you're not validating the input what if `item` equals `-100` or `'cat'`?